spa/spa.go

```diff
@@ -0,0 +1,48 @@
+package spa
+
+import (
+	"embed"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/NYTimes/gziphandler"
+)
+
+// Init creates a single page application manager.
+func Init(build embed.FS, dir string, index string) (*SPA, error) {
+	fsys, err := fs.Sub(build, dir)
+	if err != nil {
+		panic(fmt.Errorf("couldn't create sub filesystem: %w", err))
+	}
+
+	_, err = fsys.Open("index.html")
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil, fmt.Errorf("ui is enabled but no index.html found: %w", err)
```

Comment:
can we define these as sentinel errors so it becomes easy for user of this package to handle them?